logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/ProtoContentMapper.kt

```diff
@@ -117,9 +119,46 @@ class ProtoContentMapperImpl(
             is MessageContent.TextEdited -> packEdited(readableContent)
             is MessageContent.FailedDecryption, is MessageContent.RestrictedAsset, is MessageContent.Unknown, MessageContent.Ignored ->
                 throw IllegalArgumentException(
-                    "Unexpected message content type: $readableContent"
+                    "Unexpected message content type: ${readableContent.getType()}"
                 )
+
+            is MessageContent.Composite -> packComposite(readableContent, expectsReadConfirmation)
+
+            is MessageContent.ButtonActionConfirmation -> TODO()
+            is MessageContent.ButtonAction -> TODO()
+        }
+    }
+
+    private fun packComposite(
+        readableContent: MessageContent.Composite,
+        expectsReadConfirmation: Boolean
+    ): GenericMessage.Content.Composite {
+        val items: MutableList<Composite.Item> = mutableListOf()
+
+        readableContent.textContent?.let {
+            val text = packText(it, expectsReadConfirmation)
+            Composite.Item.Content.Text(text.value).also {
+                items.add(Composite.Item(it))
+            }
         }
+        readableContent.buttonList.map {
+            Composite.Item.Content.Button(
+                button = Button(
+                    text = it.text,
+                    id = it.id
+                )
+            )
+        }.also {
+            items.addAll(it.map { Composite.Item(it) })
+        }
+
```

Comment:
this could be one nice looking function returning a result of Composite.Item, we limit the mutable thingy to one fuciton with one responsibility.

Reply:
how about using the packText and packButton to map the text and buttons to proto ?